src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpectatorView.cs

```diff
@@ -244,6 +270,59 @@ private bool TryCreateRecordingService(out IRecordingService recordingService)
 #endif
         }
 
+        private void SetupNetworkConfigurationVisual()
+        {
+#if UNITY_ANDROID || UNITY_IOS
+            GameObject mobileNetworkConfigurationVisualPrefab = defaultMobileNetworkConfigurationVisualPrefab;
+            if (NetworkConfigurationSettings.IsInitialized && NetworkConfigurationSettings.Instance.OverrideMobileNetworkConfigurationVisualPrefab != null)
+            {
+                mobileNetworkConfigurationVisualPrefab = NetworkConfigurationSettings.Instance.OverrideMobileNetworkConfigurationVisualPrefab;
+            }
+
+            mobileNetworkConfigurationVisual = Instantiate(mobileNetworkConfigurationVisualPrefab);
+            networkConfigurationVisual = mobileNetworkConfigurationVisual.GetComponentInChildren<INetworkConfigurationVisual>(true);
+            if (networkConfigurationVisual == null)
+            {
+                Debug.LogError("Network configuration visual was not found. No connection will be established.");
+                return;
+            }
+

```

Comment:
What happens to the created prefab at this point?

Could we have a different approach, where we instantiate if it's null, then show/hide?

Reply:
hm, we probably need to destroy this visual at some point. But, having the ability to show/hide may prove beneficial on disconnect.